### ISSUE TITLE
Log event to system eventlog when switching desktops

### ISSUE THREAD

**Issue by DonatJR** [SharkCagey/HTWG_shark_cage#62]
logging of service install / uninstall is already done by the installer / system

**Comment by DonatJR** [comment]
I'm not sure it is worth it to include event logging just for the switch desktop event. The process of setting up event logging seems to be rather involved, maybe we can wait until we really need it or have identified more events that need to be logged.
Some resources:
* https://msdn.microsoft.com/en-us/library/windows/desktop/aa385780(v=vs.85).aspx
* https://docs.microsoft.com/en-gb/windows/desktop/ETW/event-tracing-portal
* https://docs.microsoft.com/en-gb/windows/desktop/tracelogging/trace-logging-portal

@SailReal If you agree, I will move this issue to the backlog

**Comment by SailReal** [comment]
Thanks for figuring out 👍.

I also think we should wait with the implementation until we have more usecases. 
So I'll put it back in the backlog.

**Comment by bencikpeter** [comment]
Wouldn´t it be worth it to implement some general logging capability into the app? all those `std::cout` calls  should IMO be replaced with some logger calls. 

Maybe some more lightweight logging framework could serve the purpose, because as @DonatJR pointed out, ETW is pretty heavyweight

**Comment by DonatJR** [comment]
@bencikpeter 
Yes, having a logging framework would be awesome! There is an issue for it in the backlog (#9). Can you recommend Log4Cplus? I have never used it but if you think it suits the needs for this project I am okay with commiting to it.
